PhysicsTools/NanoAOD/python/nano_cff.py

```diff
@@ -313,18 +317,21 @@ def nanoAOD_customizeCommon(process):
         nanoAOD_addDeepBTag_switch = cms.untracked.bool(False),
         nanoAOD_addDeepBoostedJet_switch = cms.untracked.bool(True), # will deactivate this in future miniAOD releases
         nanoAOD_addDeepDoubleX_switch = cms.untracked.bool(True), 
+        nanoAOD_addParticleNet_switch = cms.untracked.bool(True),
```

Comment:
do you need to re-run the particleNet also when is available in the mini ?
For the re-mini we can use the one stored in the miniAOD ?
For the old samples [ run2_nanoAOD_94X2016, run2_nanoAOD_94XMiniAODv1, run2_nanoAOD_94XMiniAODv2, run2_nanoAOD_102Xv1, run2_nanoAOD_106Xv1 ] indeed we need to re-run.

Reply:
> do you need to re-run the particleNet also when is available in the mini ?
No. We can use the ones stored in MiniAOD once they are available (e.g., in UL re-mini).

Reply:
ok. good the taggers are already available in master version.

--> for master:
 nanoAOD_produceParticleNet_switch = cms.untracked.bool(False)
if run2_nanoAOD_106Xv1 + anyOld:
 nanoAOD_produceParticleNet_switch = cms.untracked.bool(True)
add deepParticleNet

for 10_6:
if NOT run2_miniAOD_devel:
 nanoAOD_produceParticleNet_switch = cms.untracked.bool(True)
add ParticleNet

probably same logic should be applied to all the previous tagger

Reply:
> ok. good the taggers are already available in master version.
> 
> --> for master:
> nanoAOD_produceParticleNet_switch = cms.untracked.bool(False)
> if run2_nanoAOD_106Xv1 + anyOld:
> nanoAOD_produceParticleNet_switch = cms.untracked.bool(True)
> add deepParticleNet
> 
> for 10_6:
> if NOT run2_miniAOD_devel:
> nanoAOD_produceParticleNet_switch = cms.untracked.bool(True)
> add ParticleNet
> 
> probably same logic should be applied to all the previous tagger

Yes indeed. Shall I change this or would you prefer to do it centrally afterwards?

Reply:
should be done here and not afterwards

Reply:
> should be done here and not afterwards

Done in https://github.com/cms-sw/cmssw/pull/31096/commits/2c7b9ae502a856b3f6f75e3665c9f47e322441d4.